### ISSUE TITLE
"An error occurred during route handling or page rendering" ved søk

### ISSUE THREAD

**Issue by tordenflesk** [olaven/NRSS#15]

**Comment by olaven** [comment]
Hei @tordenflesk! Takk for at du rapporterer inn dette. Jeg tar en titt.

**Comment by olaven** [comment]
For meg ser det ut som om dette er rate-limitingen fra NRK som slår inn, og derfor ser jeg dette issuet i sammenheng med https://github.com/olaven/NRSS/issues/8#issuecomment-1468470453. Jeg har en ambisjon om å bygge om hele systemet slik at den ikke er like hard mot NRK sine servere (ideer og tanker mottas med takk, for øvrig).

Opplever du dette konsekvent, eller er det bare av og på, slik som for meg?

**Comment by timharek** [comment]
Kunne dette blitt løst med å indeksere alle podcastene semi-regelmessig med [`Deno.cron`](https://docs.deno.com/deploy/kv/manual/cron) og `Deno.KV` siden ting allerede er hosted på Deno Deploy? Da kunne søket heller vært internt/lokalt på appnivå istedenfor å alltid gjøre oppslag mot NRK

**Comment by olaven** [comment]
> Kunne dette blitt løst med å indeksere alle podcastene semi-regelmessig med [`Deno.cron`](https://docs.deno.com/deploy/kv/manual/cron) og `Deno.KV` siden ting allerede er hosted på Deno Deploy? Da kunne søket heller vært internt/lokalt på appnivå istedenfor å alltid gjøre oppslag mot NRK

Ja, det er i de baner jeg også har tenkt. Samme problematikk gjelder med RSS-feedene; de får 429 rett som det er. Der kunne man lagret en episode en gang, og kun hentet nye en gang i døgnet f.eks. 

@timharek i og med at du har vist litt interesse for prosjektet, er dette noe du kunne tenke deg å se på, alene eller sammen? I så fall kan jeg laget et issue hvor vi kan jobbe videre :) Jeg har som skrevet hatt en ambisjon om å gjøre noe slikt lenge, jeg har bare ikke prioritert det.

**Comment by timharek** [comment]
> @timharek i og med at du har vist litt interesse for prosjektet, er dette noe du kunne tenke deg å se på, alene eller sammen?

Sikkert lurt å se på sammen, jeg har ikke kjennskap til OpenAPI-filene og helt hvordan de fungerer. Tror også enhetstester hadde vært en fordel å fått på plass 😊

**Comment by espenol** [comment]
Hei! Jeg får denne feilmeldingen ved forsøk på å laste feed for en spesifikk podcast, f.eks. https://nrss.deno.dev/api/feeds/ekko

Kanskje det er et separat issue siden det ikke er ved søk?

**Comment by olaven** [comment]
@espenol takk for at du rapporterer!

Det har vært mye feil de siste dagene fordi prosjektet har brukt opp kvoten sin hos Deno. 
Jeg har oppgradert medlemskapet nå. Det kan fort bli litt dyrt, så jeg skal tenke ut et vis å få finnansiert det på som føles OK for de som bruker programmet. Forslag mottas med takk :)

**Comment by timharek** [comment]
> prosjektet har brukt opp kvoten sin hos Deno.

@olaven gjelder det for Deno.KV, båndbredde eller noe annet? 😊

**Comment by olaven** [comment]
> > prosjektet har brukt opp kvoten sin hos Deno.
> 
> @olaven gjelder det for Deno.KV, båndbredde eller noe annet? 😊

Deno.KV :) Vi kan alltids tweake balansen paa spamming av NRK vs. Deno KV. Naa lener det seg paa Deno KV reads.

**Comment by timharek** [comment]
Hmm, det er kanskje mer gunstig å self-hoste en VPS da og bruke SQLite el.? Betaler du da $20 for Pro?

Kan få en VPS hos Hetzner til under $5 per måned:
https://www.hetzner.com/cloud/

**Comment by olaven** [comment]
@timharek har du erfaring med om de $5 vil holde? For kontekst ser du trafikken her.
![Scherm­afbeelding 2024-04-24 om 11 15 19](https://github.com/olaven/nrss/assets/18209665/e3820811-a45e-4934-bb4b-a6b1c117dac8)

**Comment by timharek** [comment]
På Hetzner får du 20 TB trafikk, det vil være "plenty" til å betjene RSS-feeds. Jeg drifter min egen VPS som kjører litt forskjellig, men den har også to forskjellige Deno Fresh apper kjørende, [Hima](https://sr.ht/~timharek/hima/) og [Filmpolitiet API](https://github.com/timharek/filmpolitiet-api). Sistnevnte er en del enklere enn NRSS, men litt som går igjen.

Jeg kan ev. fyre opp NRSS på min VPS og se hvordan det er å self-hoste [Deno.KV](https://github.com/denoland/denokv). Hva tenker du @olaven?

**Comment by olaven** [comment]
> Jeg kan ev. fyre opp NRSS på min VPS og se hvordan det er å self-hoste [Deno.KV](https://github.com/denoland/denokv). Hva tenker du @olaven?

Kjør!
